R/read_ini.R

```diff
@@ -0,0 +1,191 @@
+#' Read Hector INI file into list
+#'
+#' @param file Path to INI file
+#' @param parse_timeseries Logical. If `TRUE` (default), use
+#'   [parse_timeseries()] to convert time series value names (e.g.
+#'   `"Ftalbedo[1980]"`) to a `data.frame`. If `FALSE`, use the full
+#'   string (with year included) as the key name.
+#' @param normalize_paths Logical. If `TRUE` (default), convert any
+#'   relative paths in the INI file to full, normalized paths,
+#'   following the same logic as the Hector internals (if the file
+#'   exists relative to the working directory, use that file.
+#'   Otherwise, look for the path in `base_dir`). If `FALSE`, leave
+#'   the paths exactly as written.
+#' @param base_dir Path to directory to search for any relative paths,
+#'   if `normalize_paths` is `TRUE`. Defaults to the directory in
+#'   which the INI file is stored.
+#' @return Nested list of the target object
+#' @author Alexey Shiklomanov
+#' @examples
+#' rcp45_ini <- read_ini(
+#'   system.file("input", "hector_rcp45.ini", package = "hector")
+#' )
+#' rcp45_ini$core$run_name
+#' rcp45_ini$onelineocean$enabled
+#' # This one will be a `data.frame`
+#' rcp45_ini$simpleNbox$Ftalbedo
+#' @export
+read_ini <- function(file,
```

Comment:
Why did we roll our own INI parser instead of using something like the `ini` or `ConfigParser` packages?  I get that we want to be able to interpolate in the time series data, but it seems like we could use the functionality of one of those packages to do the initial read-in and then make a pass through the structure to find any time series that needed to be read.  Is there some other functionality we're incorporating that we can't easily get from one of those packages?

Reply:
No specific reason, other than to avoid an external package dependency (or multiple recursive dependencies, in the case of `ConfigParser`) and maintain control of the parser in case the INI format or other aspects of the API change.

Neither of those packages seem like they're widely used (`ini` has one GitHub star, `ConfigParser` has none), and the task of parsing an INI file didn't seem particularly complicated, so I didn't have a good reason to trust others' implementation over our own.

Reply:
Not sure that everyone who installs and uses a package from CRAN goes to github and stars the repository.  I certainly don't.  Speaking as someone who once devised a bunch of regexps to parse a bunch of XML inputs, I've come around to using other people's parsers (though not, perhaps, if they use R6 to pretend like they're Python packages, as ConfigParser appears to do).